lib/rucio/client/commands/replica.py

```diff
@@ -0,0 +1,174 @@
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from argparse import SUPPRESS
+from typing import TYPE_CHECKING
+
+from rucio.client.commands.bin_legacy.rucio import list_dataset_replicas, list_file_replicas, list_suspicious_replicas
+from rucio.client.commands.bin_legacy.rucio_admin import declare_bad_file_replicas, declare_temporary_unavailable_replicas, list_pfns, quarantine_replicas, set_tombstone
+from rucio.client.commands.command_base import CommandBase
+from rucio.common.utils import StoreAndDeprecateWarningAction
+
+if TYPE_CHECKING:
+    from argparse import ArgumentParser
+
+    from rucio.client.commands.utils import OperationDict
+
+
+class Replica(CommandBase):
+    def module_help(self) -> str:
+        return "Interact with a Data IDentifier at a specific Rucio Service Element."
+
+    def _operations(self) -> dict[str, "OperationDict"]:
+        return {
+            "list": {"call": self.list_, "docs": "List the replicas of a DID and its PFNs.", "namespace": self.list_namespace},
+            "dataset": {"call": self.dataset, "docs": "List replica datasets", "namespace": self.dataset_namespace},
+            "pfn": {"call": self.pfn, "docs": "Show the pfn for replicas of a DID", "namespace": self.pfn_namespace},
+            "tombstone": {"call": self.tombstone, "docs": "Add a tombstone which will mark the replica as ready for deletion by a reaper daemon", "namespace": self.tombstone_namespace},
+        }
```

Comment:
Let’s go back to our basic building blocks:

* `add`
* `list`
* `remove`
* `show`
* `update`

And let’s focus on `list`: this is the one that we *have* to make work because it is a frequently-used command. `add` and `remove` we likely won’t implement (such functions exist, but it’s highly unusual to do so directly from the CLI). `show` and `update` are useful, but not as critical as `list`.

We have to implement two commands: `list-file-replicas` and `list-dataset-replicas`. It doesn’t help that the latter has a lot of complex options because it also returns the PFNs.

*I* wouldn’t mind if the output depended on the type of the DID: if a collection you get `list-dataset-replicas` and if a file you get `list-file-replicas`. But we would have to add some sort of option (e.g. `--files` or `--recursive`) so that, given a collection, you can get its file replicas.

I think this will require some unfortunate plumbing. But how acceptable would it be as an interface?

Reply:
If the output depends on the type of DID, it is possible for a user to get the right answer to the wrong question without knowing it. I have a strong distaste for these sort of contextual commands for that reason. 

Would `rucio replica list {file|dataset}` be an acceptable compromise? 

I think `rucio replica pfn` can probably be removed in favor of `rucio replica [solution to the list problem] --pfn`, and `rucio replica tombstone` can be swapped with `remove` (with the note that it's controlled by a reaper daemon) though.

Reply:
> Would `rucio replica list {file|dataset}` be an acceptable compromise?

I think I like this.